### PR TITLE
[stable10] support php7.3

### PR DESCRIPTION
**Pull request by patrickjahns**
## Description
Allow installation on php7.3 and add basic unit tests for it 

## Related Issue

- Fixes https://github.com/owncloud/core/issues/34464

## Motivation and Context
Allow to run ownCloud on php7.3

## How Has This Been Tested?

- 🤖 

We should additionally make a PR testing all database on php7.3 and run all acceptance tests with php7.3 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Technical debt
- [ ] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -698,6 +698,19 @@ matrix:
       INSTALL_SERVER: true
       INSTALL_TESTING_APP: true
 
+    # PHP 7.2
```
Comment (phil-davis) [inline .drone.yml]: s/7.2/7.3

File: index.php [hunk 2]
```diff
@@ -36,8 +36,8 @@
 }
 
 // Show warning if PHP 7.3 is used as ownCloud is not compatible with PHP 7.3
-if (\version_compare(PHP_VERSION, '7.3.0alpha1') !== -1) {
-	echo 'This version of ownCloud is not compatible with PHP 7.3<br/>';
+if (\version_compare(PHP_VERSION, '7.4.0alpha1') !== -1) {
```
Comment (DeepDiver1975) [inline index.php]: the same check is in console.php as well

**Review by DeepDiver1975 — DISMISSED** [review]

**Comment by patrickjahns** [comment]
Errors from drone relate all to `sabre/dav`:

```
1) OCA\DAV\Tests\unit\ServerTest::test with data set "principals" ('principals/users/admin', array('caldav', 'oc-resource-sharing', 'carddav'))
ErrorException: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?

/drone/src/lib/composer/sabre/dav/lib/CalDAV/ICSExportPlugin.php:321
/drone/src/apps/dav/lib/Server.php:160
/drone/src/apps/dav/tests/unit/ServerTest.php:45

2) OCA\DAV\Tests\unit\CardDAV\CardDavBackendTest::testUpdateProperties
ErrorException: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?

/drone/src/lib/composer/sabre/vobject/lib/Component/VCard.php:514
/drone/src/apps/dav/tests/unit/CardDAV/CardDavBackendTest.php:381

3) OCA\DAV\Tests\unit\CardDAV\CardDavBackendTest::testSearch with data set #0 ('John', array('FN'), array(array('uri0', 'John Doe'), array('uri1', 'John M. Doe')), 100, 0)
ErrorException: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?

/drone/src/lib/composer/sabre/vobject/lib/Component/VCard.php:514
/drone/src/apps/dav/tests/unit/CardDAV/CardDavBackendTest.php:491

4) OCA\DAV\Tests\unit\CardDAV\CardDavBackendTest::testSearch with data set #1 ('M. Doe', array('FN'), array(array('uri1', 'John M. Doe')), 100, 0)
ErrorException: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?

/drone/src/lib/composer/sabre/vobject/lib/Component/VCard.php:514
/drone/src/apps/dav/tests/unit/CardDAV/CardDavBackendTest.php:491

5) OCA\DAV\Tests\unit\CardDAV\CardDavBackendTest::testSearch with data set #2 ('Do', array('FN'), array(array('uri0', 'John Doe'), array('uri1', 'John M. Doe')), 100, 0)
ErrorException: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?

/drone/src/lib/composer/sabre/vobject/lib/Component/VCard.php:514
/drone/src/apps/dav/tests/unit/CardDAV/CardDavBackendTest.php:491

6) OCA\DAV\Tests\unit\CardDAV\CardDavBackendTest::testSearch with data set "check if duplicates are handled correctly" ('John', array('FN', 'CLOUD'), array(array('uri0', 'John Doe'), array('uri1', 'John M. Doe')), 100, 0)
ErrorException: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?

/drone/src/lib/composer/sabre/vobject/lib/Component/VCard.php:514
/drone/src/apps/dav/tests/unit/CardDAV/CardDavBackendTest.php:491

7) OCA\DAV\Tests\unit\CardDAV\CardDavBackendTest::testSearch with data set "case insensitive" ('john', array('FN'), array(array('uri0', 'John Doe'), array('uri1', 'John M. Doe')), 100, 0)
ErrorException: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?

/drone/src/lib/composer/sabre/vobject/lib/Component/VCard.php:514
/drone/src/apps/dav/tests/unit/CardDAV/CardDavBackendTest.php:491

8) OCA\DAV\Tests\unit\CardDAV\CardDavBackendTest::testSearch with data set "search limit" ('John', array('FN'), array(array('uri0', 'John Doe')), 1, 0)
ErrorException: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?

/drone/src/lib/composer/sabre/vobject/lib/Component/VCard.php:514
/drone/src/apps/dav/tests/unit/CardDAV/CardDavBackendTest.php:491

9) OCA\DAV\Tests\unit\CardDAV\CardDavBackendTest::testSearch with data set "search offset" ('John', array('FN'), array(array('uri1', 'John M. Doe')), 1, 1)
ErrorException: "continue" targeting switch is equivalent to "break". Did you mean to use "continue 2"?

/drone/src/lib/composer/sabre/vobject/lib/Component/VCard.php:514
/drone/src/apps/dav/tests/unit/CardDAV/CardDavBackendTest.php:491
```

To become 7.3 compliant we need to either fix `sabre/dav` for these parts - or need to move on and deprecate finally 5.6

**Comment by DeepDiver1975** [comment]
> To become 7.3 compliant we need to either fix `sabre/dav` for these parts - or need to move on and deprecate finally 5.6

sabre/dav will very unlikely add 7.3 support to an older version - we have to follow best practice just like any other project and deprecate php 5.6 :man_shrugging:

**Comment by PVince81** [comment]
we're now working on deprecating PHP 5.6 here https://github.com/owncloud/core/pull/34698

so this becomes unblocked but would still however require a major update of Sabre, which might or might not come with some surprises. need careful testing

**Comment by DeepDiver1975** [comment]
you will need a final sabre/dav 4.0 release ... ping me Wednesday on this ...

**Comment by MichaIng** [comment]
@patrickjahns @DeepDiver1975 
Hey guys. Is PHP7.3 support planned for v10.2.0 release or does it not make it anymore (since RC stage)?

**Comment by phil-davis** [comment]
> Hey guys. Is PHP7.3 support planned for v10.2.0 release or does it not make it anymore (since RC stage)?

10.2 does not get PHP 7.3 support. PHP 7.3 support is for "real soon after" (tm)

**Comment by MichaIng** [comment]
@phil-davis 
Thanks for the info.

**Comment by DeepDiver1975** [comment]
sabre/dav 4.0.0 is about to arrive - https://github.com/sabre-io/dav/pull/1163

**Comment by DeepDiver1975** [comment]
here we go: https://github.com/sabre-io/dav/releases/tag/4.0.0

**Comment by phil-davis** [comment]
I rebased and added cherry-pick of "Adjust public link lock tests to not expect locktoken" from #35183 - that should get acceptance tests passing.

**Review by phil-davis — APPROVED** [review]
LGTM, and I expect CI to pass now.

**Comment by phil-davis** [comment]
@patrickjahns @DeepDiver1975 please review again.
I made a test PR #35724 to run a lot of the drone jobs with PHP 7.3
- all the acceptance tests pass - good
- some phpunit jobs died when trying to generate code coverage. e.g.
https://drone.owncloud.com/owncloud/core/19097/172
ran exactly the same sqlite with PHP 7.3 10,198 unit tests as:
https://drone.owncloud.com/owncloud/core/19097/234

but with codecov enabled.

It ends with:
```
OK, but incomplete, skipped, or risky tests!
Tests: 10810, Assertions: 56504, Skipped: 206.

Generating code coverage report in Clover XML format ... done
[PHP Fatal error:  Uncaught ErrorException: Unknown: Cannot destroy the zip context: Can't remove file: No such file or directory in Unknown:0
Stack trace:
#0 [internal function]: OCA\Files\Command\Scan->exceptionErrorHandler(2, 'Unknown: Cannot...', 'Unknown', 0, NULL)
#1 {main}
  thrown in Unknown on line 0]
```

Maybe there is some codecov issue with PHP 7.3?

But that could be looked into in the future? All the tests run fine.

**Pull request merged**